js/compiler/compiler.go

```diff
@@ -44,103 +44,120 @@ var (
 		"highlightCode": false,
 	}
 
-	compilerInstance *Compiler
-	once             sync.Once
+	once        sync.Once // nolint:gochecknoglobals
+	globalBabel *babel    // nolint:gochecknoglobals
 )
 
-// A Compiler uses Babel to compile ES6 code into something ES5-compatible.
-type Compiler struct {
-	vm *goja.Runtime
+// CompatibilityMode specifies the JS compatibility mode
+// nolint:lll
+//go:generate enumer -type=CompatibilityMode -transform=snake -trimprefix CompatibilityMode -output compatibility_mode_gen.go
+type CompatibilityMode uint8
 
-	// JS pointers.
-	this      goja.Value
-	transform goja.Callable
-	mutex     sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
-}
+const (
+	// CompatibilityModeExtended achieves ES6+ compatibility with Babel and core.js
+	CompatibilityModeExtended CompatibilityMode = iota + 1
+	// CompatibilityModeBase is standard goja ES5.1+
+	CompatibilityModeBase
+)
 
-// Constructs a new compiler.
-func New() (*Compiler, error) {
-	var err error
-	once.Do(func() {
-		compilerInstance, err = new()
-	})
+// A Compiler compiles JavaScript source code (ES5.1 or ES6) into a goja.Program
+type Compiler struct{}
 
-	return compilerInstance, err
+// New returns a new Compiler
+func New() *Compiler {
+	return &Compiler{}
 }
 
-func new() (*Compiler, error) {
-	conf := rice.Config{
-		LocateOrder: []rice.LocateMethod{rice.LocateEmbedded},
+// Transform the given code into ES5
+func (c *Compiler) Transform(src, filename string) (code string, srcmap *SourceMap, err error) {
+	var b *babel
+	if b, err = newBabel(); err != nil {
+		return
 	}
 
-	babelSrc := conf.MustFindBox("lib").MustString("babel.min.js")
+	return b.Transform(src, filename)
+}
 
-	c := &Compiler{vm: goja.New()}
-	if _, err := c.vm.RunString(babelSrc); err != nil {
-		return nil, err
+// Compile the program in the given CompatibilityMode, optionally running pre and post code.
+func (c *Compiler) Compile(src, filename string, pre, post string,
```

Comment:
Nit: this could be `src, filename, pre, post string`

Reply:
Fixed in a8507425.